### PR TITLE
add xarray accessor `stage`

### PR DESCRIPTION
**Pull request by observingClouds**
This PR adds an xarray accessor `xr.Dataset().stage()` that allows the user to request data from tape without accumulating it in memory (e.g. via `xr.Dataset().load()`) or immediately doing resource intensive computations (e.g. via `xr.Dataset().compute()`).

### CODE REVIEW

**Comment by observingClouds** [comment]
The envisioned workflow is as following:

```python
import dask
import xarray as xr
import slkspec.xr_accessor
from dask_jobqueue import SLURMCluster

dask.config.config.get('distributed').get('dashboard').update({'link':'{JUPYTERHUB_SERVICE_PREFIX}/proxy/{port}/status'})

cluster = SLURMCluster(project="<PROJECT>",
                               cores=128,
                               interface='ib0',
                               walltime='00:20:00',
                               extra=["--lifetime", "15m", "--lifetime-stagger", "4m"],
                               silence_logs=verbose,
                               memory="256GB",
                               queue='compute',
                               job_extra=["--exclusive"],
                              )
client = Client(cluster)


ds=xr.open_mfdataset("slk:///arch/mh0010/m300408/showcase/dataset.zarr", engine="zarr")
ds_sel = ds.isel(time=0)
result = ds_sel.mean()  # lazy operation
result.stage()  # retrieve all chunks to disk that will be needed for the computation

cluster.scale(1)
result.compute()  # do the actual computation
cluster.scale(0)
```

The above code is run on a cheap compute resource (in case of Levante e.g. on the partition `shared` with `salloc -A <PROJECT> -p shared --mem=6GB`. Just enough resources are reserved to run the retrieval from disk, which is initialised with `ds_sel.stage()`.
After the successful staging, i.e. the data is transferred to the local disk, a powerful `compute` resource is spun up to do the actual computation.

**Comment by observingClouds** [comment]
The PR works already as described above but it still needs to be tested with larger datasets.

**Comment by observingClouds** [comment]
With the additional commits the datasets are loaded and directly discarded. Even for large datasets the required RAM remains low.

**Comment by observingClouds** [comment]
I added now also support for `xr.DataArray`s in addition to `xr.Dataset`s.

File: slkspec/xr_accessor.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+from itertools import compress
+
+import dask
+import xarray as xr
+
+
+@xr.register_dataarray_accessor("stage")
+@xr.register_dataset_accessor("stage")
```
Comment (dcherian) [inline slkspec/xr_accessor.py:8]: I strongly recommend adding a `slk` namespace or something similar

```suggestion
@xr.register_dataarray_accessor("slk")
@xr.register_dataset_accessor("slk")
```

so the usage is `DataArray.slk.stage()`. Then there's no confusion about whether `stage` is a built-in function provided by Xarray, or an external one registered using an accessor.

**Pull request merged**